pkg/services/rpcsrv/server.go

```diff
@@ -951,7 +951,10 @@ func (s *Server) calculateNetworkFee(reqParams params.Params) (any, *neorpc.Erro
 			}
 			w.InvocationScript = inv.Bytes()
 		}
-		gasConsumed, _ := s.chain.VerifyWitness(signer.Account, tx, &w, int64(s.config.MaxGasInvoke))
+		gasConsumed, err := s.chain.VerifyWitness(signer.Account, tx, &w, int64(s.config.MaxGasInvoke))
+		if err != nil && !errors.Is(err, core.ErrInvalidSignature) {
```

Comment:
I'm not sure that this change is good, because we have the case of contract-based witnesses with default parameters pushed on stack, and the initial idea of that was to return at least some fee value for such witnesses. And with this change if contract won't be satisfied with the default parameters, the result is likely to be `core.ErrVerificationFailed`, and the user will get nothing except the error. Won't it be better to leave this part as is?

Reply:
The trick is that you only get some valid `gasConsumed` in the `ErrInvalidSignature` case, otherwise it's just zero and doesn't add any value. And this makes some sense in general, if the script has returned `false` then at least it has executed correctly, but if it can't produce some boolean value, something is horribly broken here and an error is justified. All of our tests survive this change, however lots of things break without `!errors.Is(err, core.ErrInvalidSignature)` condition on this line.

Reply:
I've also thought about changing `VerifyWitness` to return some GAS in non-bool cases, but decided that it can hide errors.